tracing-subscriber/src/fmt/mod.rs

```diff
@@ -878,102 +883,104 @@ pub fn try_init() -> Result<(), impl Error + Send + Sync + 'static> {
+//         buf: &'a Mutex<Vec<u8>>,
+//     }
+
+//     impl<'a> MockWriter<'a> {
+//         pub(crate) fn new(buf: &'a Mutex<Vec<u8>>) -> Self {
+//             Self { buf }
+//         }
+
+//         pub(crate) fn map_error<Guard>(err: TryLockError<Guard>) -> io::Error {
+//             match err {
+//                 TryLockError::WouldBlock => io::Error::from(io::ErrorKind::WouldBlock),
+//                 TryLockError::Poisoned(_) => io::Error::from(io::ErrorKind::Other),
+//             }
+//         }
+
+//         pub(crate) fn buf(&self) -> io::Result<MutexGuard<'a, Vec<u8>>> {
+//             self.buf.try_lock().map_err(Self::map_error)
+//         }
+//     }
+
+//     impl<'a> io::Write for MockWriter<'a> {
+//         fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+//             self.buf()?.write(buf)
+//         }
+
+//         fn flush(&mut self) -> io::Result<()> {
+//             self.buf()?.flush()
+//         }
+//     }
+
+//     pub(crate) struct MockMakeWriter<'a> {
+//         buf: &'a Mutex<Vec<u8>>,
+//     }
+
+//     impl<'a> MockMakeWriter<'a> {
+//         pub(crate) fn new(buf: &'a Mutex<Vec<u8>>) -> Self {
+//             Self { buf }
+//         }
+//     }
+
+//     impl<'a> MakeWriter for MockMakeWriter<'a> {
+//         type Writer = MockWriter<'a>;
+
+//         fn make_writer(&self) -> Self::Writer {
+//             MockWriter::new(self.buf)
+//         }
+//     }
+
+//     #[test]
+//     fn impls() {
+//         let f = format::Format::default().with_timer(time::Uptime::default());
+//         let subscriber = Subscriber::builder().on_event(f).finish();
+//         let _dispatch = Dispatch::new(subscriber);
+
+//         let f = format::Format::default();
+//         let subscriber = Subscriber::builder().on_event(f).finish();
+//         let _dispatch = Dispatch::new(subscriber);
+
+//         let f = format::Format::default().compact();
+//         let subscriber = Subscriber::builder().on_event(f).finish();
+//         let _dispatch = Dispatch::new(subscriber);
+//     }
+
+//     #[test]
+//     fn subscriber_downcasts() {
+//         let subscriber = Subscriber::new();
+//         let dispatch = Dispatch::new(subscriber);
+//         assert!(dispatch.downcast_ref::<Subscriber>().is_some());
+//     }
+
+//     #[test]
+//     fn subscriber_downcasts_to_parts() {
+//         let subscriber = Subscriber::builder().finish();
+//         let dispatch = Dispatch::new(subscriber);
+//         assert!(dispatch.downcast_ref::<format::DefaultFields>().is_some());
+//         assert!(dispatch.downcast_ref::<LevelFilter>().is_some());
+//         assert!(dispatch.downcast_ref::<format::Format>().is_some())
+//     }
+
+//     #[test]
+//     #[cfg(feature = "registry_unstable")]
+//     fn is_lookup_meta() {
+//         fn assert_lookup_meta<T: crate::registry::LookupMetadata>(_: T) {}
+//         assert_lookup_meta(Subscriber::builder().finish())
+//     }
+// }
```

Comment:
I think we should be able to reimplement all these APIs (all the builder options, etc) using the new code internally. Can we do that, and remove the commented-out code, before we merge this?